tests/test_result.py

```diff
@@ -5,6 +5,18 @@
 from qibo.result import CircuitResult, MeasurementOutcomes, load_result
 
 
+def test_measurementoutcomes_probabilties(backend):
+    c = models.Circuit(3)
+    c.add(gates.H(0))
+    c.add(gates.M(0, 2))
+    probabilities = c().probabilities(qubits=[0, 2])
+    c.has_collapse = True
+    repeated_probabilities = c().probabilities()
+    result = MeasurementOutcomes(c.measurements, backend, probabilities=probabilities)
+    backend.assert_allclose(probabilities, repeated_probabilities, atol=1e-1)
```

Comment:
In order to increase the tolerance here, you can either use more shots so that the sampled probabilities converge more to the exact ones, or use a circuit where the outcomes are fixed (non-probabilistic), for example using X gates instead of H.